torchdynamo/variables/user_defined.py

```diff
@@ -190,6 +191,17 @@ def call_method(
                 assert not kwargs
                 return self.odict_getitem(tx, args[0])
 
+            if (
+                name == "__contains__"
+                and isinstance(self.value, frozenset)
+                and len(args) > 0
+                and isinstance(args[0], TorchVariable)
+            ):
+                # This handles contains frozenset contains checks for PyTorch
+                # ops.
+                # TODO(future PR): more general support for frozenset
+                return ConstantVariable(args[0].value in self.value, **options)
```

Comment:
is it fine to not guard since frozenset is immutable?

Reply:
For UserDefindObjectVariable we only guard on TYPE_MATCH:
https://github.com/pytorch/torchdynamo/blob/f21fd574dd7b07e3959103106c88bdfe27ca9e8c/torchdynamo/variables/builder.py#L295

So if you have:
```
funcs1 = frozenset([torch.add])
funcs2 = frozenset([torch.sub])
```

Both of those will pass the guards for one another, because they have the same type.  

I think the best thing to do here would be to not map set/frozenset to UserDefindObjectVariable.  For frozenset, we could just map it to a ConstantVariable -- but we will need to check all the contents of the set are also constants.

A more general solution would be to map them to new type SetVariable, which would be similar to ConstDictVariable.